lib/validation.rb

```diff
@@ -234,12 +234,12 @@ def self.validate_url(url)
     fail ValidationFailed.new({url: "Invalid URL"})
   end
 
-  def self.validate_core_quota(project, resource_type, requested_core_count)
-    if !project.quota_available?(resource_type, requested_core_count)
-      current_used_core_count = project.current_resource_usage(resource_type)
+  def self.validate_cpu_quota(project, resource_type, requested_cpu_count)
+    if !project.quota_available?(resource_type, requested_cpu_count)
+      current_used_cpu_count = project.current_resource_usage(resource_type)
       effective_quota_value = project.effective_quota_value(resource_type)
 
-      fail ValidationFailed.new({size: "Insufficient quota for requested size. Requested core count: #{requested_core_count}, currently used core count: #{current_used_core_count}, maximum allowed core count: #{effective_quota_value}, remaining core count: #{effective_quota_value - current_used_core_count}"})
+      fail ValidationFailed.new({size: "Insufficient quota for requested size. Requested CPU count: #{requested_cpu_count}, currently used CPU count: #{current_used_cpu_count}, maximum allowed CPU count: #{effective_quota_value}, remaining CPU count: #{effective_quota_value - current_used_cpu_count}"})
```

Comment:
If I remember correctly, @fdr thought we should stick to using `core` and `vCPU`. The `CPU` was making things more complicated. Why did we decide to use `CPU` instead of `vCPU`?

Reply:
well vCPU is definitely unambiguously correct, it means one thread state regardless of how it is multiplexed onto hardware. We could make it a house rule that `CPU` = `vCPU`, I don't have any opinion about that.

Note that some of the newer Xeon chips (Sierra Forest) on x64 also do away with SMT, so life gets even harder or three years from now. But the conceit of the patch overall is basically right: my dogmatic adherence to cores only holds for when you want to compare _dedicated_ compute resources, eliminating the 2x noise from SMT that doesn't result in 2x performance, to help people compare things (e.g. an arm-1 roughly equal to a x64-1, now we have the problem all the other providers have: arm-2 is better compared to x64-4)

Reply:
Anyway, this is my stance on core vs vcpu, for which I have thrown in the towel ages ago now:

![image](https://github.com/user-attachments/assets/ff09f6e8-b4d8-48c5-a271-e39a7cc91208)